bricks/flutter_clean_base/__brick__/{{project_name.snakeCase()}}/lib/core/arch/provider/base_provider.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:{{project_name}}/core/arch/domain/entity/failure/failure.dart';
+
+abstract class BaseProvider<S> extends ChangeNotifier {
+  @protected
+  late StreamController<Failure> _errorStreamController;
+
+  @protected
+  late StreamController<bool> _progressStreamController;
+
+  Stream<Failure> get failureStream => _errorStreamController.stream;
```

Comment:
@cozvtieg9, perhaps we should add a mixin for failure and progress logic to the project?